### PR TITLE
fix logging in tools/internal

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
fix error logging in tools/internal utilities

###### Change Log  

- fix error logging in tools/internal utilities

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #[49082611](https://microsoft.visualstudio.com/OS/_workitems/edit/49082611)

###### Test Methodology

- Local build

### CODE REVIEW

File: toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go [hunk 1]
```diff
@@ -764,7 +762,7 @@ func (r *RpmRepoCloner) refreshPackagesCache() (err error) {
 
 	stdout, stderr, err := shell.Execute("tdnf", args...)
 	if err != nil {
-		logger.Log.Errorf("Failed to run 'tdnf makecache'. Stdout:\n%s\nStderr:\n%s\nError: %s.", stdout, stderr, err)
+		return fmt.Errorf("failed to run 'tdnf makecache': \n%s\n%s\n%w", stdout, stderr, err)
```
Comment (cwize1) [inline toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go:765]: Concatenating the stdout and stderr together without any annotation could be a little confusing. Also, I wouldn’t necessarily add the stdout and stderr to the error message without having a good idea of what the output could contain. We want the error message to be simple and readable.

File: toolkit/tools/internal/pkggraph/pkggraph.go [hunk 2]
```diff
@@ -442,7 +442,7 @@ func (g *PkgGraph) CreateCollapsedNode(versionedPkg *pkgjson.PackageVer, parentN
 	defer func() {
 		// graph manipulation calls may panic on error (such as duplicate node IDs)
 		if r := recover(); r != nil {
-			err = fmt.Errorf("collapsing nodes (%v) into (%s) failed, error: %s", nodesToCollapse, versionedPkg, r)
+			err = fmt.Errorf("failed to collapse nodes (%v) into (%s) failed:\n%w\n%s", nodesToCollapse, versionedPkg, err, r)
```
Comment (cwize1) [inline toolkit/tools/internal/pkggraph/pkggraph.go:445]: If you are adding a cleanup error to an original error, you should clearly mark the cleanup err within the error message.

**Review by cwize1 — CHANGES_REQUESTED** [review]

File: toolkit/tools/internal/pkggraph/pkggraph.go (hunk 2 above)
Comment (cwize1) [inline toolkit/tools/internal/pkggraph/pkggraph.go:445]: If there was a panic, will `err` contain a value?

File: toolkit/tools/internal/pkggraph/pkggraph.go [hunk 3]
```diff
@@ -454,7 +454,7 @@ func (g *PkgGraph) CreateCollapsedNode(versionedPkg *pkgjson.PackageVer, parentN
 			for _, node := range nodesToCollapse {
 				lookupErr := g.addToLookup(node, false)
 				if lookupErr != nil {
-					logger.Log.Errorf("Failed to add node (%s) back to lookup table. Error: %s", node.FriendlyName(), lookupErr)
+					err = fmt.Errorf("failed to add node (%s) back to lookup table:\n%w\n%w", node.FriendlyName(), lookupErr, err)
```
Comment (cwize1) [inline toolkit/tools/internal/pkggraph/pkggraph.go:457]: When dealing with cleanup errors, you can either:

1. Add the cleanup error to the existing error (if any).
2. Just report the cleanup error as a warning.

Both are acceptable. However, if you are modifying existing code, it might be easier to use Option 2.

File: toolkit/tools/internal/pkggraph/pkggraph.go (hunk 2 above)
Comment (neha170) [reply toolkit/tools/internal/pkggraph/pkggraph.go:445]: Thanks for pointing this out. defer() would either catch `err` or `panic`, so will modify this to return only the required values.

File: toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go [hunk 4]
```diff
@@ -764,7 +762,8 @@ func (r *RpmRepoCloner) refreshPackagesCache() (err error) {
 
 	stdout, stderr, err := shell.Execute("tdnf", args...)
 	if err != nil {
-		logger.Log.Errorf("Failed to run 'tdnf makecache'. Stdout:\n%s\nStderr:\n%s\nError: %s.", stdout, stderr, err)
+		logger.Log.Errorf("Failed to run 'tdnf makecache'\nstdout:\n%v", stdout)
```
Comment (cwize1) [inline toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go:765]: Why did you remove the `stderr` from the logs?

File: toolkit/tools/internal/pkggraph/pkggraph.go [hunk 5]
```diff
@@ -1056,8 +1053,8 @@ func (g *PkgGraph) AddMetaNode(from []*PkgNode, to []*PkgNode) (metaNode *PkgNod
 			for _, n := range to {
 				toNames = fmt.Sprintf("%s %s", toNames, n.FriendlyName())
 			}
-			logger.Log.Errorf("Couldn't add meta node from [%s] to [%s]", fromNames, toNames)
-			logger.Log.Panicf("Adding meta node failed.")
+			err := fmt.Errorf("failed to add meta node from (%s) to (%s)", fromNames, toNames)
+			logger.Log.Panicf("Adding meta node failed:\n%v\n%v", err, r)
```
Comment (cwize1) [inline toolkit/tools/internal/pkggraph/pkggraph.go:1057]: Why bother creating the `err` object? Why not just add everything to the `logger.Log.Panicf()`?

File: toolkit/tools/internal/pkggraph/pkggraph.go [hunk 6]
```diff
@@ -442,7 +442,7 @@ func (g *PkgGraph) CreateCollapsedNode(versionedPkg *pkgjson.PackageVer, parentN
 	defer func() {
 		// graph manipulation calls may panic on error (such as duplicate node IDs)
 		if r := recover(); r != nil {
-			err = fmt.Errorf("collapsing nodes (%v) into (%s) failed, error: %s", nodesToCollapse, versionedPkg, r)
+			err = fmt.Errorf("failed to collapse nodes (%v) into (%s):\n%s", nodesToCollapse, versionedPkg, r)
```
Comment (cwize1) [inline toolkit/tools/internal/pkggraph/pkggraph.go:445]: Use `%w` for `r`.

File: toolkit/tools/internal/pkggraph/pkggraph.go [hunk 7]
```diff
@@ -454,7 +454,7 @@ func (g *PkgGraph) CreateCollapsedNode(versionedPkg *pkgjson.PackageVer, parentN
 			for _, node := range nodesToCollapse {
 				lookupErr := g.addToLookup(node, false)
 				if lookupErr != nil {
-					logger.Log.Errorf("Failed to add node (%s) back to lookup table. Error: %s", node.FriendlyName(), lookupErr)
+					err = fmt.Errorf("%w\nfailed to add node (%s) back to lookup table:\nlookupErr:\n%w", err, node.FriendlyName(), lookupErr)
```
Comment (cwize1) [inline toolkit/tools/internal/pkggraph/pkggraph.go:457]: Perhaps?

```golang
err = fmt.Errorf("%w\ncleanup error: failed to add node (%s) back to lookup table:\n%w", err, node.FriendlyName(), lookupErr)
```

**Review by cwize1 — CHANGES_REQUESTED** [review]

File: toolkit/tools/internal/pkggraph/pkggraph.go (hunk 6 above)
Comment (neha170) [reply toolkit/tools/internal/pkggraph/pkggraph.go:445]: Golang wants to use `%v` or `%s`

**Review by cwize1 — APPROVED** [review]
